src/Common/FitOnFhir.Common/Services/AuthStateService.cs

```diff
@@ -0,0 +1,90 @@
+{
+    public class AuthStateService : IAuthStateService
+    {
+        private readonly AuthenticationConfiguration _authenticationConfiguration;
+        private readonly IJwtSecurityTokenHandlerProvider _jwtSecurityTokenHandlerProvider;
+        private readonly ILogger _logger;
+
+        public AuthStateService(
+            AuthenticationConfiguration authenticationConfiguration,
+            IJwtSecurityTokenHandlerProvider jwtSecurityTokenHandlerProvider,
+            ILogger<AuthStateService> logger)
+        {
+            _authenticationConfiguration = EnsureArg.IsNotNull(authenticationConfiguration, nameof(authenticationConfiguration));
+            _jwtSecurityTokenHandlerProvider = EnsureArg.IsNotNull(jwtSecurityTokenHandlerProvider, nameof(jwtSecurityTokenHandlerProvider));
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+        }
+
+        public string ExternalIdentifier { get; set; }
+
+        public string ExternalSystem { get; set; }
+
+        /// <inheritdoc/>
+        public AuthState CreateAuthState(HttpRequest httpRequest)
+        {
+            var externalId = HttpUtility.UrlDecode(EnsureArg.IsNotNullOrWhiteSpace(
+                httpRequest.Query[Constants.ExternalIdQueryParameter], $"query.{Constants.ExternalIdQueryParameter}"));
+
+            var externalSystem = HttpUtility.UrlDecode(EnsureArg.IsNotNullOrWhiteSpace(
+                httpRequest.Query[Constants.ExternalSystemQueryParameter], $"query.{Constants.ExternalSystemQueryParameter}"));
+
+            // is this for anonymous logins?
+            if (_authenticationConfiguration.IsAnonymousLoginEnabled)
+            {
+                ExternalIdentifier = externalId;
+                ExternalSystem = externalSystem;
+            }
+            else
+            {
+                // do not allow the ExternalId or ExternalSystem query params when authentication is enabled
+                if (!string.IsNullOrEmpty(externalId) || !string.IsNullOrEmpty(externalSystem))
+                {
+                    _logger.LogError($"{Constants.ExternalIdQueryParameter} and {Constants.ExternalSystemQueryParameter} are forbidden query parameters with non-anonymous authorization.");
+                    throw new ArgumentException();
+                }
+
+                // extract the token from the header.
+                if (!httpRequest.TryGetTokenStringFromAuthorizationHeader(JwtBearerDefaults.AuthenticationScheme, out string token))
+                {
+                    _logger.LogError("The request Authorization header is invalid.");
+                    return default;
+                }
+
+                // Read the token into the handler.
+                _jwtSecurityTokenHandlerProvider.SetMapInboundClaims(false);
+                JwtSecurityToken jwtSecurityToken;
+                try
+                {
+                    jwtSecurityToken = _jwtSecurityTokenHandlerProvider.ReadJwtToken(token);
```

Comment:
Since this try catch is written in 2 places, can we just add this logic to the IJwtSecurityTokenHandlerProvider?